monkey/monkey_island/cc/resources/events.py

```diff
@@ -11,10 +14,26 @@
 class Events(AbstractResource):
     urls = ["/api/events"]
 
+    def __init__(
+        self,
+        agent_event_queue: IAgentEventQueue,
+        event_serializer_registry: EventSerializerRegistry,
+    ):
+        self._agent_event_queue = agent_event_queue
+        self._event_serializer_registry = event_serializer_registry
+
     # Agents needs this
     def post(self):
-        event = request.json
+        events = request.json
+
+        for event in events:
+            try:
+                serializer = self._event_serializer_registry[event["type"]]
+                deserialized_event = serializer.deserialize(event)
+            except (TypeError, ValidationError) as err:
```

Comment:
What raises a `ValidationError`? `deserialize()` should raise `TypeError` or `ValueError`. I'll update the docstring on `IEventSerializer` on develop.